client/js/libs/handlebars/ircmessageparser/parseStyle.js

```diff
@@ -107,25 +143,19 @@ const properties = ["bold", "textColor", "bgColor", "italic", "underline", "reve
 
 function prepare(text) {
 	return parseStyle(text)
-		.filter(fragment => fragment.text.length)
-		.reduce((prev, curr, i) => {
-			if (i === 0) {
-				return prev.concat([curr]);
-			}
-
-			const lastEntry = prev[prev.length - 1];
-			if (properties.some(key => curr[key] !== lastEntry[key])) {
-				return prev.concat([curr]);
+		// This optimizes fragments by combining them together when all their values
+		// for the properties defined above are equal.
+		.reduce((prev, curr) => {
+			if (prev.length) {
+				const lastEntry = prev[prev.length - 1];
+				if (properties.every(key => curr[key] === lastEntry[key])) {
+					lastEntry.text += curr.text;
+					lastEntry.end += curr.text.length;
+					return prev;
+				}
 			}
-
-			lastEntry.text += curr.text;
-			return prev;
-		}, [])
-		.map((fragment, i, array) => {
-			fragment.start = i === 0 ? 0 : array[i - 1].end;
-			fragment.end = fragment.start + fragment.text.length;
-			return fragment;
-		});
```

Comment:
I have made this `.map` unnecessary by adding `start`/`end` information as part of the `emitFragment()` function (see above) and summing up `end` in the optimization step above.